src/modules/auth/strategies/google.strategy.ts

```diff
@@ -0,0 +1,39 @@
+import { PassportStrategy } from '@nestjs/passport';
+import { Strategy, VerifyCallback } from 'passport-google-oauth20';
+import { Injectable } from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
+import { EnvConfig } from '../../../shared/configs/env.config';
+
+@Injectable()
+export class GoogleStrategy extends PassportStrategy(Strategy, 'google') {
+  constructor(private configService: ConfigService<EnvConfig, true>) {
+    super({
+      clientID: configService.get('GOOGLE_CLIENT_ID', {
+        infer: true,
+      }),
+      clientSecret: configService.get('GOOGLE_CLIENT_SECRET', {
+        infer: true,
+      }),
+      callbackURL: configService.get('GOOGLE_CALLBACK_URL', { infer: true }),
+      scope: ['email', 'profile'],
+    });
+  }
+
+  async validate(
+    accessToken: string,
+    refreshToken: string,
+    profile: any,
+    done: VerifyCallback,
+  ): Promise<any> {
+    const { name, emails, photos, id } = profile;
+    const user = {
+      email: emails[0].value,
+      firstName: name.givenName,
+      lastName: name.familyName,
+      picture: photos[0].value,
+      accessToken,
+      googleId: id,
+    };
```

Comment:
Add missing value handling. Name, email, and picture might not be available depending on Google account privacy settings. Use optional chaining.